### PR TITLE
Added links to databases to pathway view title

### LINKED ISSUE #930: Long text in title extends beyond bounds of app-bar-title container
## Environment info

- Browser version (client) : Chrome 67.0.3396.99 (Official Build) (64-bit)
- Node.js version (server) : n/a

## Current (buggy) behaviour

Long text in titles will extend beyond the app-bar-title container bounds. 

## Desired behaviour

Not extend outside of container.

## Minimum steps to reproduce

See [TP53 regulates transcription of additional cell cycle genes whose exact role in the p53 pathway remain uncertain | Reactome ](http://localhost:3000/pathways?uri=http%3A%2F%2Fidentifiers.org%2Freactome%2FR-HSA-6804115)

image

### LINKED ISSUE #1018: Link to pathway and interaction database providers
## Description of new feature

Provide, when available, a link to the pathway or interaction database provider's site.

Specifically, in the pathway title bar. Not sure if there are other instances.

image

## Motivation for new feature

Enhance the navigation to and from data provider sites.

### PR DESCRIPTION
**Pull request by MichaelWrana**
Fixes #1018 
Fixes #930

Reversed the order of `Pathway name | Datasource` => `Datasource | Pathway name`.
Removed "Unknown Title" text if it were to appear.

The map should be correct for all datasources as of right now, but there is also a string similarity calculation included as backup in case the names change in future.

screen shot 2018-08-20 at 12 00 23 pm
screen shot 2018-08-20 at 12 00 38 pm
screen shot 2018-08-20 at 12 00 43 pm

### CODE REVIEW

**Comment by jvwong** [comment]
Just curious where did you get the URLs for the databases?

**Comment by MichaelWrana** [comment]
Googled it for most of them.  If I couldn't find a direct link, the link is to the journal article for the database (ex: pid goes to https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2686461/)

**Comment by jvwong** [comment]
OK, I'm fine with the decision for those with no live URL.

**Comment by MichaelWrana** [comment]
Update: Turns out this doesn't actually fix #930

**Comment by maxkfranz** [comment]
How about using `:` instead of `|`?  With the DB name on the left, it looks like the title of the site beside the PC logo.  Maybe a bit more spacing between the DB name and the logo too.

**Comment by d2fong** [comment]
Also, maybe we can make all links consistent and use the plain link style that the tooltips and the protein info box uses

**Comment by MichaelWrana** [comment]
Something like this?

screen shot 2018-08-20 at 10 36 32 am

**Comment by maxkfranz** [comment]
Yes, but I think Dylan meant just the blue underline on links.

**Comment by d2fong** [comment]
There should be a ```plain-link``` class included by the base css file we use.  I think you can just use that.

File: src/client/features/pathways/index.js [hunk 1]
```diff
@@ -84,9 +86,32 @@ class Pathways extends React.Component {
     }
   }
 
+  baseNameToUrl(baseName){
```
Comment (d2fong) [inline src/client/features/pathways/index.js]: I think this function belongs in datasource-url-map

File: src/client/features/pathways/index.js [hunk 2]
```diff
@@ -84,9 +86,32 @@ class Pathways extends React.Component {
...
+    return databaseURL;
+  }
+
   render() {
     let { loading, pathway, cySrv, activeMenu, networkEmpty } = this.state;
 
+    const baseDatasource = pathway.datasource();
+    const databaseURL = this.baseNameToUrl(baseDatasource);
+    let baseName = pathway.name();
+    if(baseName.toLowerCase() === 'unknown title'){
```
Comment (d2fong) [inline src/client/features/pathways/index.js]: Maybe you can just change the default value returned by pathway.name().  What do you think?

Comment (MichaelWrana) [reply src/client/features/pathways/index.js]: I can't figure out where `pathway.name()` is being defined, do you know?

Comment (d2fong) [reply src/client/features/pathways/index.js]: https://github.com/PathwayCommons/app-ui/tree/development/src/client/models/pathway

**Comment by d2fong** [comment]
@jvwong thoughts on this?

**Comment by jvwong** [comment]
Works for me!

**Comment by d2fong** [comment]
Thoughts on the way that the name and title are displayed?

**Comment by jvwong** [comment]
Actually, just a sec: Those database names aren't going to work: e.g. pid

**Comment by jvwong** [comment]
'pid' is the only one - probably should be displayed as 'NCI-PID'

**Comment by MichaelWrana** [comment]
I think the names are based on metadata for the network:
In `/src/client/models/pathway`: 
```
  datasource(){
    return _.get(this.raw, 'graph.pathwayMetadata.dataSource.0', 'Unknown datasource');
  }
```

I could add in a quick fix and directly change 'pid' to 'NCI-PID', but that seems like bad practice.  Any ideas?

**Comment by jvwong** [comment]
OK then - I am fine with all of this. 

Notes for future: Datasource derived from [traversal into pathway object](https://github.com/PathwayCommons/app-ui/blob/6e2c136b5f22260335a8b1ac0e57c1c5a224a845/src/server/graph-generation/index.js#L16). Should be improved with [mapping file](https://github.com/PathwayCommons/cpath2/issues/303)

**Comment by gbader** [comment]
Hi - just noticed the move of the database name to be first.  Users don't care about the database name - they want to see the pathway name first.  Database groups, I'm sure, would like to see their name first, but they are secondary to users.  Did you guys discuss that?

**Comment by d2fong** [comment]
I think the initial proposal is in this issue, but I don't think we had any discussion about it.

https://github.com/PathwayCommons/app-ui/issues/1018

**Comment by jvwong** [comment]
That was my suggestion to put the `source: title`- some titles are huge. Anyways,  we can go with `title: datasource`.

**Pull request merged**